Editor/CourseValidation/Context/IContext.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Innoactive.CreatorEditor.CourseValidation
+{
+    /// <summary>
+    /// Context is used to indicate the position in the course structure.
+    /// </summary>
+    public interface IContext
+    {
+        /// <summary>
+        /// Parent context, can be null.
+        /// </summary>
+        IContext Parent { get; }
+
+        /// <summary>
+        /// If this is true, you can call Select on this object.
+        /// </summary>
+        bool IsSelectable { get; }
```

Comment:
You could separate Select() method to ISelectableContext interface to enforce that.